### ISSUE TITLE
Add functionality to do inverse prediction

### ISSUE THREAD

**Issue by psolymos** [psolymos/opticut#26]
Create new branch (`inverse-prediction`) and implement the following functions:

* `ipredict` generic and methods (default, opticut): currently called `calibrate`, but `ipredict` should better reflect the intention. This will only run a single instance with any ynew/xnew arguments.
* multiclass measures based on Sokolova et al.
* `calibrate` method to implement LOO. Implement comparison based on multiclass measures similarly to AIC but meybe only for 2 objects: LOTO (leave one taxon out) vs. reference (all taxa LOO)

**Comment by psolymos** [comment]
Write tmpfile for model directly to avoid multiple thread clashes.

**Comment by psolymos** [comment]
Deal with possibly different species subsets in `object` and `ynew`:

* species present in `ynew` but not in `object`: drop
* species present in `object` but not in `ynew`: 0 value is information, thus setting it to 0 is not correct. Need to subset JAGS input data to contain the intersect of species.

This is best achieved by a subset method for the opticut/uncertainty model classes: carefully subset species (modifying other attributes etc. as needed). This is useful and safe option.

**Comment by psolymos** [comment]
- [x] `dcoptions` used to avoid overwriting model file when running in parallel -- needs testing
- [x] `subset` method added and used in `ipredict.opticut`
- [x] add multiclass functions
- [x] implement a multinomial fitting function (`multicut`) modeled after `opticut` and `optilevels` (all dist, because it is already used in `rankComb`)
- [x] modify `ipredict.default` for `multicut` objects (binomial and poisson cases only).
- [ ] write documentation and add LOSO/LOTO examples in `\dontrun` or implement fast version:
  - leave-one-site-out (LOSO): opticut((n-1) x m), ipredict(1 x m), repeat n times, multiclass assessment
  - leave-one-taxon-out (TOTO): opticut(n x (m-1)), ipredict(n x (m-1)), repeat m times, assess
- [x] `predict` method for opticut and multicut objects (with args: object, xnew, gnew; all dist types)
- [x] `predict` method for multicut objects (with args: object, xnew, gnew; all dist types)
- [x] MCMC and analytic options for `ipredict`: analytic should implement additive loglik (binomial and poisson cases only)

**Comment by psolymos** [comment]
Needed methods for multicut class: 
- [x] print, 
- [x] summary, 
- [x] print.summary.multicut, 
- [x] plot, 
- [x] fitted, 
- [x] predict, 
- [x] ipredict, 
- [x] getMLE, 
- [x] strata, 
- [x] subset, 
- [x] as.data.frame, 
- [x] as.data.frame.summary.multicut.

Not sure if uncertainty and related methods need to be implemented as well. It is not needed for ipredict, stratification is fixed, and bootstrap is kind of trivial in that case.

Need to test all the dist cases (compare coef with getMLE output)

**Comment by psolymos** [comment]
Most of the prep work done for milestone v1.0. Here is the list of remaining items for milestone v1.1:

- [x] `ipredict` methods for opticut and multicut objects implementing IP.
- [x] add multiclass functions for assessing IP (branch multiclass).
- [x] implement LOSO/LOTO examples and functions for analytic and MCMC based IP (call it `ipower` referring to [posterior] indicator power).

**Comment by psolymos** [comment]
Need to figure out final object structures, print/summary/plot methods, write documentation and tests. Otherwise the multiclass branch has the new functionality in `/extras`.

**Comment by psolymos** [comment]
kappa stat needs to be fully fledged out, also considering prior weights.
